cli/ast.rs

```diff
@@ -372,6 +392,82 @@ pub fn parse(
   })
 }
 
+/// A low level function which transpiles a source module into an swc
+/// SourceFile.
+pub fn transpile_module(
+  filename: &str,
+  src: &str,
+  media_type: &MediaType,
+  emit_options: &EmitOptions,
+  cm: Rc<SourceMap>,
+) -> Result<(Rc<SourceFile>, Module), AnyError> {
+  let comments = SingleThreadedComments::default();
+  let syntax = get_syntax(media_type);
+  let source_file =
+    cm.new_source_file(FileName::Custom(filename.to_string()), src.to_string());
+  let lexer = Lexer::new(
+    syntax,
+    TARGET,
+    StringInput::from(&*source_file),
+    Some(&comments),
+  );
+  let mut parser = swc_ecmascript::parser::Parser::new_from(lexer);
+  // TODO(@kitsonk) this seems to be a problem with the type of result
+  // from `.parse_module` that cannot be coerced.
+  let module = match parser.parse_module() {
+    Ok(m) => m,
+    Err(err) => bail!("Parsing failed: {:?}", err),
+  };
```

Comment:
Need to be mapped to `DiagnosticBuffer` like in `parse` function - why isn't `parse` function from this module used in this function?

Reply:
Thanks for noticing that, I took a look at it, and it would take a decent amount of refactoring of things to get to the point where we could use `parse()` (or share code between `transpile_module()` and `parse()`), mainly because `parse()` currently doesn't share a source map between invocations (but maybe it should?) where as the bundling example from kdy that this is based on did.  There is the emit side of it too which shares a lot with the other emits...

I made a note of the refactoring opportunities in there, but it isn't totally straight forward, especially without impacting other parts of the code.